Cutscenes.py

```diff
@@ -485,33 +634,1313 @@ def patch_cutscenes(rom: Rom, songs_as_items: bool, settings: Settings) -> None:
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[3].headers[0].actor_list.actors[11],
+        # GTG Rotating Statue Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[4].headers[0].actor_list.actors[3],
+        # GTG Megaton Statue Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[5].headers[0].actor_list.actors[18],
+        # GTG Lava Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[6].headers[0].actor_list.actors[17],
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[6].headers[0].actor_list.actors[18],
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[6].headers[0].actor_list.actors[19],
+        # GTG Dinolfos Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[7].headers[0].actor_list.actors[14],
+        # GTG Ice Arrow Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[8].headers[0].actor_list.actors[4],
+        # GTG Shellblade Room
+        scenes[SceneIDs.GERUDO_TRAINING_GROUND].rooms[9].headers[0].actor_list.actors[22],
+        # Death Mountain Crater
+        scenes[SceneIDs.DEATH_MOUNTAIN_CRATER].rooms[1].headers[2].actor_list.actors[44],
+        # Thieves' Hideout Green Cell Room 3 torches
+        scenes[SceneIDs.THIEVES_HIDEOUT].rooms[1].headers[0].actor_list.actors[9],
+        # Thieves' Hideout Red Cell Room 1 torch
+        scenes[SceneIDs.THIEVES_HIDEOUT].rooms[2].headers[0].actor_list.actors[9],
+        # Thieves' Hideout Green Cell Room 4 torches
+        scenes[SceneIDs.THIEVES_HIDEOUT].rooms[4].headers[0].actor_list.actors[11],
+        # Thieves' Hideout Blue Cell Room 2 torches
+        scenes[SceneIDs.THIEVES_HIDEOUT].rooms[5].headers[0].actor_list.actors[14],
     ]
-    for address in wonder_talk2_y_coordinates:
-        rom.write_byte(address, 0xFB)
+
+    # Pre-scene-framework hack replaced the first byte with 0xFB
+    bit_mask1 = int.from_bytes(b'\xFB\x00', 'big', signed=True)
+    bit_mask2 = int.from_bytes(b'\xFB\xFF', 'big', signed=True)
+    for actor in wonder_talk2_actor_entries:
+        actor.pos.y = (actor.pos.y | bit_mask1) & bit_mask2
 
     if 'frogs2' in settings.misc_hints:
         # Prevent setting the replaced textbox flag so that the hint is easily repeatible by walking over the spot again.
         # And move the hint spot down the log so that it doesn't pop every time a song is played, and let some room to do ocarina item glitch.
-        rom.write_int16s(0x2059412, [0x03C0, 0x00E2, 0xFAA6]) # Move coordinates. Original value : 1000, 205, -1202. New value : 960, 226, -1370.
-        rom.write_byte(0x205941F, 0xBF) # Never set the flag.
+        actor = scenes[SceneIDs.ZORAS_RIVER].rooms[0].headers[0].actor_list.actors[58]
+        actor.pos = Vec3s(960, 226, -1370) # original 1000, -205, -1202
+        actor.params = 0x4BBF # original 0x4BBB
+
+
+# Gaps in IDs are intentional
+# https://github.com/zeldaret/oot/blob/7235af2249843fb68740111b70089bad827a4730/include/z64cutscene.h#L35-L165
+class CutsceneCommandID(IntEnum):
```

Comment:
man, rando sure is a hot mess when it comes to organization. 

I find it kind of odd to just stuff these important type definitions after a bunch of function patching routines split out of patches.py. I think we need to consider moving this type access stuff into a new folder like "z64types" or something